src/target/riscv/riscv.c

```diff
@@ -3425,16 +3422,11 @@ static int riscv_rw_memory(struct target *target, const riscv_mem_access_args_t
 		return ERROR_OK;
 	}
 
-	int mmu_enabled;
-	int result = riscv_mmu(target, &mmu_enabled);
-	if (result != ERROR_OK)
-		return result;
-
 	RISCV_INFO(r);
-	if (!mmu_enabled)
+	if (riscv_virt2phys_mode_is_hw(target))
 		return r->access_memory(target, args);
```

Comment:
Do I understand correctly that when virt2phys is off, we should also directly call `r->access_memory`?

```suggestion
	if (riscv_virt2phys_mode_is_off(target))
		/* No address translation is performed. Virtual == physical is assumed. */
		return r->access_memory(target, args);
	if (riscv_virt2phys_mode_is_hw(target))
		/* Address translation virtual -> physical will be performed by the hardware. */	
		return r->access_memory(target, args);

	/* Otherwise address translation is performed by OpenOCD - for each 4 KiB page separately.
	 * The memory operation must be split to page-sized chunks.
	 */
	
	...
```

Reply:
@fk-sc It looks like you have independently came to the same conclusion [in this comment](https://github.com/riscv-collab/riscv-openocd/pull/1256/files#r2093662876).